src/base-input.js

```diff
@@ -163,7 +164,8 @@ export default class BaseInput extends LitElement {
       maxLength:               { type: Number },
       minLength:               { type: Number },
       showPassword:            { state: true },
-      setCustomValidity:       { type: String }
+      setCustomValidity:       { type: String },
+      keyValidate:             { type: Boolean }
```

Comment:
I think we could name this better, especially since it's no longer firing on `keydown`. What about `validateBeforeBlur` if we want to be explicit? Also open to other suggestions.

Reply:
I chose this name as the feature we have directly mimics the standard [HTMLObjectElement.setCustomValidity](https://developer.mozilla.org/en-US/docs/Web/API/HTMLObjectElement/setCustomValidity)

This attribute accepts a string and sets a custom message. The comment about using `validateBeforeBlur`, I think you are getting this confused with the `keyValidate` attribute.

Reply:
I was suggesting we rename `keyValidate` to `validateBeforeBlur` since that's more descriptive.

Reply:
Ahhhhhh... I see what I did there. I was hyperfocused on the `Validity` word there. 

`validateBeforeBlur` feels awkwardly procedural. Why not say what it is? `validateOnInput` is more direct.

Reply:
`validateOnInput` works for me 👍